shop/src/main/java/ro/msg/learning/shop/service/impl/ProductServiceImpl.java

```diff
@@ -0,0 +1,104 @@
+package ro.msg.learning.shop.service.impl;
+
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+import ro.msg.learning.shop.exceptions.ProductNotFoundException;
+import ro.msg.learning.shop.model.Product;
+import ro.msg.learning.shop.model.ProductCategory;
+import ro.msg.learning.shop.model.Supplier;
+import ro.msg.learning.shop.repository.ProductRepository;
+import ro.msg.learning.shop.service.ProductCategoryService;
+import ro.msg.learning.shop.service.ProductService;
+import ro.msg.learning.shop.service.SupplierService;
+import ro.msg.learning.shop.utils.Status;
+
+import java.util.Collection;
+import java.util.Optional;
+
+@Service
+@RequiredArgsConstructor
+public class ProductServiceImpl implements ProductService {
+
+    private final ProductRepository productRepository;
+    private final ProductCategoryService productCategoryService;
+    private final SupplierService supplierService;
+
+    @Override
+    public Status addProduct(Product product) {
+        Optional<Supplier> supplier=supplierService.findByName(product.getSupplier().getName());
+        if(!supplier.isPresent()){
+            return Status.SUPPLIER_NOT_FOUND;
+        }
+        Optional<ProductCategory> productCategory=productCategoryService.findByName(product.getCategory().getName());
+        if(!productCategory.isPresent()){
+            return Status.CATEGORY_NOT_FOUND;
+        }
+        product.setSupplier(supplier.get());
+        product.setCategory(productCategory.get());
+        System.out.println(product);
```

Comment:
hello darkness my old friend 👾 . Remove printing in console. Not safe, not clean